include/rmm/mr/device/per_device_resource.hpp

```diff
@@ -121,6 +129,25 @@ RMM_EXPORT inline auto& get_map()
   return device_id_to_resource;
 }
 
+/**
+ * @briefreturn{Reference to the lock}
+ */
+inline std::mutex& ref_map_lock()
+{
+  static std::mutex ref_map_lock;
+  return ref_map_lock;
+}
+
+// This symbol must have default visibility, see: https://github.com/rapidsai/rmm/issues/826
+/**
+ * @briefreturn{Reference to the map from device id -> resource_ref}
+ */
+RMM_EXPORT inline auto& get_ref_map()
```

Comment:
*question*: I see that `initial_resource` also returns a `static` object (a `device_memory_resource *`), but that one is not `RMM_EXPORT`. I think it is benign because two `cuda_memory_resource` objects are completely interchangeable. But in theory I think it suffers from the same problem as mentioned in that issue.

Reply:
Fixed.